### ISSUE TITLE
`datatype` of `gr.Dataframe` cannot be updated

### ISSUE THREAD

**Issue by hysts** [gradio-app/gradio#3732]
### Describe the bug

It seems `gr.Dataframe.update()` doesn't support `datatype`. I want to choose column names of a dataframe and hide the selected columns, but I don't think it's possible without it.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

```python
import gradio as gr
import pandas as pd

COLUMN_NAMES = [
    'A',
    'B',
    'C',
]
COLUMN_DATATYPES = [
    'markdown',
    'str',
    'number',
]

data = [['# aaa', 'bbb', 123]]
df = pd.DataFrame(data, columns=COLUMN_NAMES)


def update_table(column_indices: list[int]) -> dict:
    columns = [COLUMN_NAMES[i] for i in column_indices]
    datatypes = [COLUMN_DATATYPES[i] for i in column_indices]
    return gr.update(value=df.loc[:, columns], datatype=datatypes)


with gr.Blocks(css='style.css') as demo:
    column_selection = gr.CheckboxGroup(label='Columns',
                                        choices=COLUMN_NAMES,
                                        value=COLUMN_NAMES,
                                        type='index')
    dataframe = gr.Dataframe(value=df,
                             datatype=COLUMN_DATATYPES,
                             type='pandas')

    column_selection.change(fn=update_table,
                            inputs=column_selection,
                            outputs=dataframe)

demo.queue(api_open=False).launch()
```

https://huggingface.co/spaces/hysts-debug/gradio-issue-3732

### Logs

```shell
3c9a3chgsg 2023-04-02T14:12:57.567Z Running on local URL:  http://0.0.0.0:7860
3c9a3chgsg 2023-04-02T14:12:57.567Z 
3c9a3chgsg 2023-04-02T14:12:57.567Z To create a public link, set `share=True` in `launch()`.
3c9a3chgsg 2023-04-02T14:13:47.521Z Traceback (most recent call last):
3c9a3chgsg 2023-04-02T14:13:47.521Z   File "/home/user/.local/lib/python3.10/site-packages/gradio/routes.py", line 393, in run_predict
3c9a3chgsg 2023-04-02T14:13:47.521Z     output = await app.get_blocks().process_api(
3c9a3chgsg 2023-04-02T14:13:47.521Z   File "/home/user/.local/lib/python3.10/site-packages/gradio/blocks.py", line 1111, in process_api
3c9a3chgsg 2023-04-02T14:13:47.521Z     data = self.postprocess_data(fn_index, result["prediction"], state)
3c9a3chgsg 2023-04-02T14:13:47.521Z   File "/home/user/.local/lib/python3.10/site-packages/gradio/blocks.py", line 1036, in postprocess_data
3c9a3chgsg 2023-04-02T14:13:47.521Z     prediction_value = postprocess_update_dict(
3c9a3chgsg 2023-04-02T14:13:47.521Z   File "/home/user/.local/lib/python3.10/site-packages/gradio/blocks.py", line 421, in postprocess_update_dict
3c9a3chgsg 2023-04-02T14:13:47.521Z     update_dict = block.get_specific_update(update_dict)
3c9a3chgsg 2023-04-02T14:13:47.521Z   File "/home/user/.local/lib/python3.10/site-packages/gradio/blocks.py", line 306, in get_specific_update
3c9a3chgsg 2023-04-02T14:13:47.521Z     specific_update = cls.update(**generic_update)
3c9a3chgsg 2023-04-02T14:13:47.521Z TypeError: Dataframe.update() got an unexpected keyword argument 'datatype'
```

### System Info

```shell
gradio==3.24.1
```

### Severity

annoying

**Comment by abidlabs** [comment]
Thanks @hysts, yes this is essentially a special case of #2358, so I'll close this issue. We'll look into this as part of the work towards 4.0!

**Comment by hysts** [comment]
@abidlabs 
Looks like #2358 was closed, but I still don't know how to update `datatype`.
The same error still occurs with `gradio==3.39.0`. Space: https://huggingface.co/spaces/hysts-debug/gradio-issue-3732-gradio-3-39-0

When columns have different `datatype`, without updating `datatype`, columns may not be shown as expected.

For example, when column `A` is `markdown` and `B` is `str`, the table is shown like this,
![](https://github.com/gradio-app/gradio/assets/25161192/5e1a689b-e8fc-4404-a216-151996e2adea)
but when hiding the column `A` without updating `datatype`, it becomes like this:
![](https://github.com/gradio-app/gradio/assets/25161192/b5729120-970d-421e-a50f-91cd43112528)
The column `B` is supposed to be shown as `str`, but as the `datatype` of `gr.Dataframe` is not changed, now the `datatype` of the column `B` became `markdown`.

**Comment by abidlabs** [comment]
Reopening, though I think this will get solved once we solve #1566, cc @aliabid94

**Comment by abidlabs** [comment]
This now works on `main` @hysts! 

Although Dataframe is kinda broken because of #5592 -- we'll get that fixed separately!
